### ISSUE TITLE
Contain semantic type output from hostile metadata

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#4613]
## Summary

`type --table` and `type --shape` can emit raw Unicode format controls extracted from valid metadata, bypassing the scanner/extraction containment boundary.

Part of #3635.

## Reproduction

A repository-owned immutable package fixture will carry a valid SRM-generated metadata assembly with focused specimens. Against its extracted companion assembly:

```bash
dotnet run --project src/dotnet-inspect -c Release -- \
  type --library MetadataConfusion.dll --all --table > type-table.txt

dotnet run --project src/dotnet-inspect -c Release -- \
  type --library MetadataConfusion.dll --all --shape > type-shape.md
```

Both outputs currently contain raw format controls:

- U+200B ZERO WIDTH SPACE in a namespace;
- U+2060 WORD JOINER in a generic-parameter name.

The same fixture should prove the intended boundary elsewhere:

- `mdi` refuses by default and spells the values under `--show-untrusted-text`;
- `library -v:d` spells hostile assembly, custom-attribute, P/Invoke, and resource values;
- a homoglyph AssemblyRef remains graphic text and is reported separately as identifier confusion.

## Expected

Semantic type extraction should preserve raw values for identity and lookup, then project every artifact-derived output value to `InertString` before table, Markdown, JSON, or browser DTO sinks unwrap it. Homoglyph review remains separate from visual containment.

Add the repository-owned real-artifact canary after its GitHub Packages fixture is published, plus close benign cases proving ordinary non-ASCII identifiers remain unchanged.
